base/docs/basedocs.jl

```diff
@@ -1977,57 +2006,151 @@ Boolean type, containing the values `true` and `false`.
+Binary format is 1 sign, 8 exponent, 23 fraction bits.
+
+The exponent for scientific notation should be entered as lower-case `f`,
+thus `2f3 === 2.0f0 * 10^3 === Float32(2_000)`.
+For arrays literals and comprehensions, the element type can be specified before
+the square brackets: `Float32[1,4,9] == Float32[i^2 for i in 1:3]`.
+
+See also [`Inf32`](@ref), [`NaN32`](@ref), [`Float16`](@ref), [`exponent`](@ref), [`frexp`](@ref).
+"""
+Float32
+
+"""
+    Float16 <: AbstractFloat <: Real
+
+16-bit floating point number type (IEEE 754 standard).
+Binary format is 1 sign, 5 exponent, 10 fraction bits.
+"""
+Float16
 
 for bit in (8, 16, 32, 64, 128)
+    Sys.WORD_SIZE == bit && continue  # Int & UInt have separate, more detailed, descriptions
+    type = Symbol(:Int, bit)
+    srange = bit > 31 ? "" : "Represents numbers `n ∈ " * repr(eval(:(typemin($type):typemax($type)))) * "`.\n"
+    unshow = repr(eval(Symbol(:UInt, bit))(bit-1))
+
     @eval begin
         """
-            Int$($bit) <: Signed
+            Int$($bit) <: Signed <: Integer
 
         $($bit)-bit signed integer type.
+
+        $($(srange))Note that such integers overflow without warning,
+        thus `typemax($($type)) + $($type)(1) < 0`.
+
+        See also [`Int`](@ref $Int), [`widen`](@ref), [`BigInt`](@ref).
         """
         $(Symbol("Int", bit))
 
         """
-            UInt$($bit) <: Unsigned
+            UInt$($bit) <: Unsigned <: Integer
 
         $($bit)-bit unsigned integer type.
+
+        Printed in hexadecimal, thus $($(unshow)) == $($(bit-1)).
         """
         $(Symbol("UInt", bit))
     end
 end
 
+@eval begin
+    """
+        $Int
+
+    $(Sys.WORD_SIZE)-bit signed integer type, `$Int <: Signed <: Integer <: Real`.
+
+    This is the default type of most integer literals, on computers for which `Sys.WORDSIZE == $(Sys.WORD_SIZE)`,
+    and has an alias `Int`. It is the type returned by functions such as [`length`](@ref), and the
+    standard type for indexing arrays. The default (and the alias) may be 32 or 64 bits on different
+    computers, indicated by `Sys.WORD_SIZE`.
+
+    Note that integers overflow without warning, thus `typemax(Int) + 1 < 0` and `10^19 < 0`.
+    Overflow can be avoided by using [`BigInt`](@ref).
+    Very large integer literals will use a wider type, for instance `10_000_000_000_000_000_000 isa Int128`.
+
+    Integer division is [`div`](@ref) alias `÷`,
+    whereas [`/`](@ref) acting on integers returns [`Float64`](@ref).
+
+    See also [`$(Symbol("Int", 96 - Sys.WORD_SIZE))`](@ref), [`widen`](@ref), [`typemax`](@ref), [`bitstring`](@ref).
+    """
+    $(Symbol(Int))
+
+    """
+        $UInt
+
+    $(Sys.WORD_SIZE)-bit unsigned integer type, `$UInt <: Unsigned <: Integer`.
+
+    Like [`Int`](@ref $Int), the alias `UInt` may point to either `UInt32` or `UInt64`,
+    according to the value of `Sys.WORD_SIZE` on a given computer.
+
+    Printed in hexadecimal, thus `UInt(15) === $(repr(UInt(15)))`.
```

Comment:
This holds because they are parsed in hexadecimal, not because they are printed in hexadecimal.

Reply:
```suggestion
    Printed and parsed in hexadecimal: `UInt(15) === $(repr(UInt(15)))`.
```